src/Service/AuthenticationServiceInterface.php

```diff
@@ -12,5 +12,6 @@
 
 interface AuthenticationServiceInterface extends GraphQLiteAuthenticationServiceInterface
 {
-    public function createToken(string $username, string $password, int $shopid = null): Token;
+    public function createAuthenticatedToken(string $username, string $password): Token;
+    public function createUnauthenticatedToken(string $username, string $usergroup = null): Token;
```

Comment:
I'd like to discuss this method @mtkoltan  I have two reasons I don't like it:

- this looks like it's only needed for the developer module, so i think it should go there
- we do not have "unauthenticated" tokens in the app, every token is tightly coupled to a user

What would you think about moving this method to the graphql developer module, name it something like `createDeveloperToken` and let it extend the `AuthenticationService`? We would then need to make the `getInitializedTokenBuilder` protected and may add a `@internal` annotaion.

Reply:
as discussed I will implement the suggestions in this module